docker/Dockerfile.base

```diff
@@ -0,0 +1,251 @@
+        #   https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#L34
+        #   https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#L114
+        dpkg-dev        \
+    # Download the source tarball and verify its integrity.
+    # https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#L56
+    && wget -O python.tar.xz "https://www.python.org/ftp/python/$PYTHON3_VERSION/Python-$PYTHON3_VERSION.tar.xz" \
+    && wget -O python.tar.xz.asc "https://www.python.org/ftp/python/$PYTHON3_VERSION/Python-$PYTHON3_VERSION.tar.xz.asc" \
+	&& GNUPGHOME="$(mktemp -d)" \
+    && export GNUPGHOME \
+	&& gpg --recv-keys "$PYTHON_SOURCE_GPG" \
+	&& gpg --verify python.tar.xz.asc python.tar.xz \
+	&& command -v gpgconf > /dev/null && gpgconf --kill all || : \
+	&& rm -rf "$GNUPGHOME" python.tar.xz.asc \
+    # Compile and install Python from source.
+    # https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#LL56C2-L99C3
+    && mkdir -p /usr/src/python \
+	&& tar --extract --directory /usr/src/python --strip-components=1 --file python.tar.xz \
+	&& rm python.tar.xz \
+    && cd /usr/src/python \
+    && gnuArch="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)" \
+    && ./configure \
+        --build="$gnuArch" \
+        --enable-loadable-sqlite-extensions \
+		--enable-optimizations \
+		--enable-option-checking=fatal \
+		--enable-shared \
+		--with-lto \
+		--with-system-expat \
+		--without-ensurepip \
+    && nproc="$(nproc)" \
+	&& LDFLAGS="-Wl,--strip-all" \
+	&& make -j "$nproc" \
+		"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+		"LDFLAGS=${LDFLAGS:-}" \
+		"PROFILE_TASK=${PROFILE_TASK:-}" \
+    && rm python \
+	&& make -j "$nproc" \
+		"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+		"LDFLAGS=${LDFLAGS:--Wl},-rpath='\$\$ORIGIN/../lib'" \
+		"PROFILE_TASK=${PROFILE_TASK:-}" \
+		python \
+	&& make install \
+	&& cd / \
+	&& rm -rf /usr/src/python \
+    # Clean up test and cache files.
+    # https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#LL100C2-L108C3
+	&& find /usr/local -depth \
+		\( \
+			\( -type d -a \( -name test -o -name tests -o -name idle_test \) \) \
+			-o \( -type f -a \( -name '*.pyc' -o -name '*.pyo' -o -name 'libpython*.a' \) \) \
+		\) -exec rm -rf '{}' + \
+	&& ldconfig \
+    && python3 --version \
+    # Create useful symlinks for python related tools.
+    # https://github.com/docker-library/python/blob/f568f56f28fab0fe87b34db777e2c2861cef002b/3.11/slim-buster/Dockerfile#L124
+	&& for src in idle3 pydoc3 python3 python3-config; do \
+		dst="$(echo "$src" | tr -d 3)"; \
+		[ -s "/usr/local/bin/$src" ]; \
+		[ ! -e "/usr/local/bin/$dst" ]; \
+		ln -svT "$src" "/usr/local/bin/$dst"; \
+	done \
```

Comment:
```suggestion
	&& for src in idle3 pydoc3 python3 python3-config; do \
		dst="$(echo "$src" | tr -d 3)"; \
		[ -s "/usr/local/bin/$src" ]; \
		[ ! -e "/usr/local/bin/$dst" ]; \
		ln -svT "$src" "/usr/local/bin/$dst"; \
	   done \
```

Reply:
I think GitHub UI display this part a bit weirdly.
This is the original text:
```dockerfile
    && for src in idle3 pydoc3 python3 python3-config; do \
        dst="$(echo "$src" | tr -d 3)"; \
        [ -s "/usr/local/bin/$src" ]; \
        [ ! -e "/usr/local/bin/$dst" ]; \
        ln -svT "$src" "/usr/local/bin/$dst"; \
    done \
```
So I think if we want `done` to align with `for` , `done` must be moved 3 spaces to the left, which is not the usual 4 spaces indent :thinking:

Reply:
It's very likely that tab and spaces are mixed in this Dockerfile. That's why GitHub UI display some parts differently. I will have a look at this.

Reply:
Yes, that is true. I will also convert all tabs into 4 spaces in the next commit.